UIElements/setSprocketsVertical.py

```diff
@@ -12,73 +12,109 @@ class SetSprocketsVertical(Widget):
     def LeftCW(self):
         degValue = float(self.data.config.get('Advanced Settings',"gearTeeth"))*float(self.data.config.get('Advanced Settings',"chainPitch"))/360.0;
         self.data.gcode_queue.put("G91 ")
-        self.data.gcode_queue.put("B09 L"+str(degValue)+" ")
+        if self.data.config.get('Advanced Settings', 'chainOverSprocket') == 'Top':
+            self.data.gcode_queue.put("B09 L"+str(degValue)+" ")
+        else:
+            self.data.gcode_queue.put("B09 L-"+str(degValue)+" ")
         self.data.gcode_queue.put("G90 ")
 
     def LeftCCW(self):
         degValue = float(self.data.config.get('Advanced Settings',"gearTeeth"))*float(self.data.config.get('Advanced Settings',"chainPitch"))/360.0;
         self.data.gcode_queue.put("G91 ")
-        self.data.gcode_queue.put("B09 L-"+str(degValue)+" ")
+        if self.data.config.get('Advanced Settings', 'chainOverSprocket') == 'Top':
+            self.data.gcode_queue.put("B09 L-"+str(degValue)+" ")
+        else:
+            self.data.gcode_queue.put("B09 L"+str(degValue)+" ")
         self.data.gcode_queue.put("G90 ")
 
     def RightCW(self):
         degValue = float(self.data.config.get('Advanced Settings',"gearTeeth"))*float(self.data.config.get('Advanced Settings',"chainPitch"))/360.0;
         self.data.gcode_queue.put("G91 ")
-        self.data.gcode_queue.put("B09 R-"+str(degValue)+" ")
+        if self.data.config.get('Advanced Settings', 'chainOverSprocket') == 'Top':
```

Comment:
This is really beautifully done. Its simple, but I like that it just works based on the settings and doesn't ask the user anything